browser/resources/brave_new_tab_page_refresh/components/context/app_provider.tsx

```diff
@@ -0,0 +1,44 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+
+import { AppModel } from '../../models/app_model'
+import { Locale } from '../../models/locale_strings'
+import { LocaleProvider } from './locale_context'
+import { NewTabProvider } from './new_tab_context'
+import { BackgroundProvider } from './background_context'
+import { SearchProvider } from './search_context'
+import { TopSitesProvider } from './top_sites_context'
+import { VPNProvider } from './vpn_context'
+import { RewardsProvider } from './rewards_context'
+
+interface Props {
+  locale: Locale
+  model: AppModel
+  children: React.ReactNode
+}
+
+// Provides all context required by the application.
+export function AppProvider(props: Props) {
```

Comment:
Note that in this iteration, we are still joining the separate model interfaces together into a single `AppModel`. An advantage of this approach is that it creates a single place for the app to perform initialization. I'll explore some alternatives as well.

Reply:
See https://github.com/brave/brave-core/commit/dcbda57453284e7b48c316a14606e79e85f66dc4 for an approach that builds upon this change and removes `AppModel`.